strawberryfields/backends/bosonicbackend/bosoniccircuit.py

```diff
@@ -803,6 +804,58 @@ def measure_dyne(self, covmat, modes, shots=1):
 
         return vals
 
+    def measure_fock(self, modes, shots=1):
+        mu = self.means
+        mean = self.get_mean_xp()
+        cov = self.get_covmat_xp()
+        weights = self.weights
+        
+        x_idxs = np.array(modes)
+        p_idxs = (x_idxs + len(np.transpose(mu))/2).astype(int)
+        modes_idxs = np.concatenate([x_idxs, p_idxs])
+        measurements = np.zeros((len(mu)))
+        ##measure photon number in every gaussian mode
+        for i in range(len(cov)):
+            cov_i = cov[i]
+            mean_i = mean[i]
+            reduced_cov_i = cov_i[np.ix_(modes_idxs, modes_idxs)]
+            reduced_mean_i = mean_i[modes_idxs]
+            if np.allclose(mu, np.zeros_like(mu)):
+                haf_i = hafnian_sample_state(reduced_cov_i, shots)
+                measurements[i] = haf_i[0][0] 
+            else:
+                haf_i = hafnian_sample_state(reduced_cov_i, shots, mean=reduced_mean_i)
+                measurements[i] = haf_i[0][0] 
+        index = np.array([np.random.choice(list(range(len(measurements))),1,p=np.abs(weights))])
+        samples = measurements[index] 
+        return samples
+    
+    def measure_threshold(self, modes, shots=1):
```

Comment:
I just checked and for the example I shared the `modes` array is of size 1, with the correct mode in it.

Reply:
sure, but you are still checking the fidelity of the whole state to the multimode vacuum state. You want the fidelity to the single mode vacuum in the mode that is being measured.

Reply:
@nquesada If I run `vacuum_fidelity = np.abs(self.fidelity_vacuum(modes))` instead, if a "no click" occurs (in this case force `measurement = 0` on a `sf.ops.Catstate(alpha=1.4)` circuit), the following state running through measure_threshold is now returning true to `self.is_vacuum()`, even if it's a `sf.ops.Catstate(alpha=1.4)` as well, with the circuit I sent. This is initially why I didn't pass the parameter because it seems to be the case for every circuit that getting a "no click" causes the next mode to be identified as vacuum. It really seems like adjusting the weight is what is causing issues - if I have two identical catstates and I calculate `vacuum_fidelity`, and multiply `self.weights` by its inverse, and calculate the `vacuum_fidelity` again, it gives 1.

Reply:
@nquesada I also ran into another problem when trying to run the tests - `chop_in_blocks_vector_multi` in ~`\strawberryfields\strawberryfields\backends\bosonicbackend\ops.py` has issues taking a single-mode circuit. I think it should be adjusted to handle the case where `id_to_keep = []`, similarly to how `chop_in_blocks_multi` returns an empty array for A.

Reply:
Thanks for finding this interesting edge case @jeykelly . There are two possible solutions: one would be two "fix" the `chop_in_blocks_vector_multi` functions the other is to us an if statement and check for the special cases where the total number of modes is 1. In this case you just need to decide whether to click or not to click and then regardless of the outcomes set the covs to be a single identity matrix, the means to be a single vector of zeros and the coefficients to be a single coefficient with value 1. I suggest you implement the second option, but first make sure that you do pass the argument `modes` to the vacuum fidelity method. That way you don't need the first `if` statement in your `measure_threshold` method.